src/libostree/ostree-repo.c

```diff
@@ -374,31 +386,47 @@ push_repo_lock (OstreeRepo          *self,
   repo_lock_info (lock, &info);
   g_debug ("Push lock: state=%s, depth=%u", info.name, info.len);
 
-  if (info.state == LOCK_EX)
+  int next_state;
+  guint *counter;
+  if (lock_type == OSTREE_REPO_LOCK_EXCLUSIVE)
     {
-      g_debug ("Repo already locked exclusively, extending stack");
-      g_queue_push_head (&lock->stack, GINT_TO_POINTER (LOCK_EX));
+      next_state = LOCK_EX;
```

Comment:
They're actually not identical because `flags` can contain `LOCK_NB`. I suppose I could use separate constants rather than reusing the flock operation flags. I kinda wanted to add `OSTREE_REPO_LOCK_UNLOCKED` to `OstreeRepoLockType` and use those constants for the state tracking, but then I don't think that's really appropriate in the API since it's not something you could pass to the any of the locking functions.

Reply:
`flags` only contains `LOCK_NB` in `pop_repo_lock()`. This comment is about `push_repo_lock()`.

Reply:
It does (it has to be passed through to the actual open fd/flock call):
```
  int flags = (lock_type == OSTREE_REPO_LOCK_EXCLUSIVE) ? LOCK_EX : LOCK_SH;
  if (!blocking)
    flags |= LOCK_NB;
```
But you're right that the assignments could be more localized so they're relationship is clearer.

Reply:
I put the assignment to `next_state` in this block so it's clearer how it's derived and how it's different from `flags`.